lib/link.ts

```diff
@@ -271,9 +286,67 @@ export abstract class Link extends Entity {
         return resolve();
       }
     });
-    log[this.type]("[%s] %s '%s' has been closed, now closing it's amqp session '%s'.",
-      this.connection.id, this.type, this.name, this.session.id);
-    return this._session.close();
+
+    if (options.closeSession) {
+      log[this.type]("[%s] %s '%s' has been closed, now closing it's amqp session '%s'.",
+        this.connection.id, this.type, this.name, this.session.id);
+      return this._session.close();
+    }
+  }
+
+  /**
+   * **It is the synchronous version of `close` where the user can call `closeSync` and not**
+   * **worry about errors caused while closing the link (fire and forget).**
+   *
+   * Closes the underlying amqp link and optionally the session as well in rhea if open.
+   * Also removes all the event handlers added in the rhea-promise library on the link
+   * and optionally it's session.
+   */
+  closeSync(options?: LinkCloseOptions): void {
```

Comment:
@GaikwadPratik - In your scenario, if you may not care about the errors that occur while closing the link then this method could be useful to you. The errors will be logged if the logger is enabled. However no error will be thrown.

Reply:
So you are saying, I should invoke _sync_ version rather than _async_ version?

Reply:
Yes you can *if you do not care about the errors that may occur while closing the link*. I added this flavor for close because many times what will the user do if an error occurs while closing something? Because you were done with this entity hence you wanted to close it. Hence this flavor can be useful in majority of scenarios.

**Note: Before using the _`sync`_ version, it would be great if you can use the latest drop and provide me the updated logs. That will help us find out the root cause of the original issue and fix it correctly.**